ocaml/tests/test_psr.ml

```diff
@@ -0,0 +1,122 @@
+let checkpoint = ref None
+let backups = ref None
+let default_ps = "default_ps"
+let new_ps = "new_ps"
+
+(* we don't make the distinction between a slave & master here.
+   it is intended that whilst the master orchestrates the procedure
+   in the real implementation, the way the master should tell itself
+   to update the PS is no different to the way the master should tell
+   a slave *)
+module Impl = struct
+  type ps = string
+  type pss = (ps * ps)
+  type fistpoint = No_fp
+                 | Accept_new_ps
```

Comment:
The module uses an acronym (psr) and `ps` is used here, too. Looking from the outside it might be good to spell out `pool_secret` somewhere.

Reply:
i mentioned the abbreviations in the mli file, but you are right - this is an indication i should change the name of the type